### PR TITLE
QA-120: Acceptance tests python3 for mender-convert 

### PR DESCRIPTION
**Pull request by lluiscampos**
Missing fixes from PR #849. See individual commits.

### CODE REVIEW

File: tests/acceptance/common.py [hunk 1]
```diff
@@ -268,51 +268,53 @@ def common_boot_from_internal(conn):
...
-        output = subprocess.check_output(["sh", "-c", "ls -t %s/%s/*%s | grep -v data*%s| head -n 1" % (builddir, sdimg_location, extension, extension)])
+        output = subprocess.check_output(["sh", "-c", "ls -t %s/*%s | grep -v data*%s| head -n 1" % (builddir, extension, extension)])
     else:
         output = subprocess.check_output(["sh", "-c", "ls -t %s/tmp*/deploy/images/*/*%s | grep -v data*%s| head -n 1" % (builddir, extension, extension)])
     output = output.decode().rstrip('\r\n')
-    print("Found latest image of type '%s' to be: %s" % (extension, output))
-    return output
+    path_to_image = os.path.abspath(output)
+    print("Found latest image of type '%s' to be: %s" % (extension, path_to_image))
+    return path_to_image
```
Comment (lluiscampos) [inline tests/acceptance/common.py]: @pasinskim FYI this is the change that makes the tests fail. Probably the working dir is not what I assumed to be. I don't have more time today but I'll look into it tomorrow.

File: tests/acceptance/common.py [hunk 2]
```diff
@@ -268,51 +268,53 @@ def common_boot_from_internal(conn):
...
+
     if configuration.get("conversion"):
-        output = subprocess.check_output(["sh", "-c", "ls -t %s/%s/*%s | grep -v data*%s| head -n 1" % (builddir, sdimg_location, extension, extension)])
+        output = subprocess.check_output(["sh", "-c", "ls -t %s/*%s | grep -v data*%s| head -n 1" % (builddir, extension, extension)])
     else:
         output = subprocess.check_output(["sh", "-c", "ls -t %s/tmp*/deploy/images/*/*%s | grep -v data*%s| head -n 1" % (builddir, extension, extension)])
     output = output.decode().rstrip('\r\n')
-    print("Found latest image of type '%s' to be: %s" % (extension, output))
-    return output
+    path_to_image = os.path.abspath(output)
```
Comment (pasinskim) [inline tests/acceptance/common.py]: Just a slight detail, but in this case. maybe we can remove above `rstrip('\r\n')` call?

File: tests/acceptance/common.py [hunk 3]
```diff
@@ -268,51 +268,53 @@ def common_boot_from_internal(conn):
...
         export_only_expr = "?"
     matcher = re.compile('^(?:export )%s([A-Za-z][^=]*)="(.*)"$' % export_only_expr)
     ret = {}
-    while True:
-        line = ps.stdout.readline()
-        if not line:
-            break
-        line = line.decode().strip()
+    for line in lines:
+        line = line.strip()
```
Comment (pasinskim) [inline tests/acceptance/common.py]: Can't we strip lines as we read those here: https://github.com/mendersoftware/meta-mender/pull/864/files#diff-1341c7e04ba8ae1f7491966f6793b12dR302

Comment (lluiscampos) [reply tests/acceptance/common.py]: We would then need to strip there and in the config line parsing `lines = config.readlines()`. It is quite harmless so I'll leave it.

File: tests/acceptance/common.py (hunk 2 above)
Comment (lluiscampos) [reply tests/acceptance/common.py]: I have reworked this change.

File: tests/acceptance/common.py (hunk 1 above)
Comment (lluiscampos) [reply tests/acceptance/common.py]: Reworked

**Comment by lluiscampos** [comment]
All acceptance tests passed after my rework. Merging!

**Pull request merged**